File: web/src/api/message.ts
import axios from 'axios';

export interface MessageData {
  id: string;
  channel_id: string;
  body: string;
  edited: boolean;
  created_at: string;
  created_by: string;
  created_by_name: string;
  created_by_avatar_url: string;
  updated_at: string;
}

export const getMessages = async (workspaceId: string, channelId: string) => {
  const response = await axios.get(`/api/v1/workspaces/${workspaceId}/channels/${channelId}/messages`, { withCredentials: true });
  return response.data as MessageData[];
};

export const createMessage = async (workspaceId: string, channelId: string, body: string) => {
  const response = await axios.post(`/api/v1/workspaces/${workspaceId}/channels/${channelId}/messages`, { body });
  return response.data as MessageData;
};

export const updateMessage = async (workspaceId: string, channelId: string, id: string, body: string) => {
  const response = await axios.put(`/api/v1/workspaces/${workspaceId}/channels/${channelId}/messages/${id}`, { body });
  return response.data as MessageData;
};

export const deleteMessage = async (workspaceId: string, channelId: string, id: string) => {
  const response = await axios.delete(`/api/v1/workspaces/${workspaceId}/channels/${channelId}/messages/${id}`);
  return response.data;
};

File: api/internal/model/message.go
package model

type MessageFilter struct {
	WorkspaceID string
	ChannelID   string
}

type Message struct {
	ID          string `json:"id"`
	WorkspaceID string `json:"workspace_id"`
	ChannelID   string `json:"channel_id"`
	Body        string `json:"body"`
	Edited      bool   `json:"edited"`
	CreatedAt   string `json:"created_at"`
	CreatedBy   string `json:"created_by"`
	UpdatedAt   string `json:"updated_at"`
	UpdatedBy   string `json:"updated_by"`
}
